### ISSUE TITLE
[Feature request] generate clean diffs with require

### ISSUE THREAD

**Issue by greg0ire** [composer/composer#3525]
ATM, the git diff I get when requiring a new package looks like this:

``` diff
already-installed/package,
- "already-installed/package-that-has-nothing-to-do-with-the-new-package": "~1.0",
+ "already-installed/package-that-has-nothing-to-do-with-the-new-package": "~1.0",
+ "new-package/i-am-requiring": "~1.0"
```

I think there are two solutions to this problem:
1. add the new package anywhere but last in the package list
2. put commas first

I think the better solution is 1.

**Comment by greg0ire** [comment]
I just learnt this was fixed by @localheinz in #3872 . Thanks a lot @localheinz !

**Comment by localheinz** [comment]
@greg0ire 

You're welcome, using this feature all the time! :wink:
